compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -301,4 +343,27 @@ object Completion {
     val Import: Mode = new Mode(4) | Term | Type
   }
 
+   /** A scope that tracks renames of the entered symbols.
+    *  Useful for providing completions for renamed symbols
+    *  in the REPL and the IDE.
+    */
+   private class RenameAwareScope extends Scopes.MutableScope {
+     private[this] val nameToSymbols: mutable.Map[Name, List[Symbol]] = mutable.Map.empty
+
+     /** Enter the symbol `sym` in this scope, recording a potential renaming. */
+     def enter[T <: Symbol](sym: T, name: Name)(implicit ctx: Context): T = {
+       nameToSymbols += name -> (sym :: nameToSymbols.getOrElse(name, Nil))
+       newScopeEntry(name, sym)
+       sym
+     }
+
+     /** Lists the symbols in this scope along with the name associated with them. */
+     def toListWithNames(implicit ctx: Context): List[(Symbol, Name)] = {
```

Comment:
Hmm, why return a `List[(Symbol, Name)]` when `getCompletions` will convert it into a `List[Name, List[Symbol]]` which is almost what `nameToSymbols` already is ?

Reply:
Were you suggesting to return directly the `Map[Name, List[Symbol]]` here? That's what I did.